api/src/model/request/play_asset_pack_fetch_request.gd

```diff
@@ -0,0 +1,118 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# Request object that handles asynchronous logic related to fetch_asset_pack().
+#
+# This object provides relevant getters so that it is possible to retrieve
+# the updated states from this object upon every state update.
+#
+# ##############################################################################
+class_name PlayAssetPackFetchRequest
+extends PlayAssetDeliveryRequest
+
+# -----------------------------------------------------------------------------
+# Emits request_completed(did_succeed, pack_name, result, exception) signal 
+# upon request succeeds(reaching CANCELED/COMPLETED)/fails(reaching FAILED).
+# 	did_succeed : boolean indicating request succeeded/failed
+# 	pack_name: String, name of the requested asset pack
+# 	result : PlayAssetPackState object
+#	exception: PlayAssetPackException object if Plugin encountered an exception
+# while handling this request
+#
+# Note: when calling fetch_asset_pack() on a non-existent pack_name, 
+# did_succeed will be false and state's status will be INVALID_REQUEST
+# -----------------------------------------------------------------------------
+signal request_completed(did_succeed, pack_name, result, exception)
+
+var _pack_name : String
+var _did_succeed : bool
+var _state : PlayAssetPackState
+var _error : PlayAssetPackException
+
+func _init(pack_name):
+	_pack_name = pack_name
+	
+	# _state will be defaulted with status of UNKNOWN
+	var default_pack_dict = {
```

Comment:
Ideally this should be const, but GDScript can not instantiate const Dictionary that contain enum values.